hydrolib/core/io/fnm/serializer.py

```diff
@@ -0,0 +1,194 @@
+        {}    *  89. Sacramento HIS file              
+        {}    *  90. rwzi HIS file                                                        O
+        {}    *  91. Industry HIS file                                                    O
+        {}    *  92. CTRL.INI                                                             I
+        {}    *  93. CAPSIM input file                                                    I
+        {}    *  94. CAPSIM input file                                                    I
+        {}    *  95. CAPSIM message file                                                  O
+        {}    *  96. CAPSIM debug file                                                    O
+        {}    *  97. Restart file na 1 uur                                                O
+        {}    *  98. Restart file na 12 uur                                               O
+        {}    *  99. Ready                                                                O
+        {}    * 100. NWRW detailed areas                                                  O
+        {}    * 101. Link flows                                                           O
+        {}    * 102. Modflow-RR                                                           O
+        {}    * 103. RR-Modflow                                                           O
+        {}    * 104. RR-balance for WLM              
+        {}    * 105. Sacramento ASCII output              
+        {}    * 106. Additional NWRW input file with DWA table                            I
+        {}    * 107. RR balans
+        {}    * 108. Kasklasse, new format                                                I
+        {}    * 109. KasInit, new format                                                  I
+        {}    * 110. KasGebr, new format                                                  I
+        {}    * 111. CropFact, new format                                                 I
+        {}    * 112. CropOW, new format                                                   I
+        {}    * 113. Soildata, new format                                                 I
+        {}    * 114. DioConfig Ini file
+        {}    * 115. Buifile voor continue berekening Reeksen
+        {}    * 116. NWRW output
+        {}    * 117. RR Routing link definitions                                          I
+        {}    * 118. Cel input file
+        {}    * 119. Cel output file
+        {}    * 120. RR Log file for Simulate
+        {}    * 121. coupling WQ salt RTC
+        {}    * 122. RR Boundary conditions file for SOBEK3
+        {}    * 123. Optional RR ASCII restart (test) for OpenDA
+        {}    * 124. Optional LGSI cachefile
+        {}    * 125. Optional meteo NetCdf timeseries inputfile rainfall
+        {}    * 126. Optional meteo NetCdf timeseries inputfile evaporation
+        {}    * 127. Optional meteo NetCdf timeseries inputfile temperature (only for RR-HBV)
+        """.format(*values))
```

Comment:
I don't like use of order, rather use the f string variant with keywords (can also be `format` though). And what happens on `None` for the optional fields? Should these be left empty `""` or will there be `None` printed?

Reply:
the `values` variable is processed see the line 
```python
values = (_get_formatted_value(v, max_len) for v in data.values())
```
which will handle None values as well as the empty space offset such that everything is nicely aligned.

f-strings could work as well, the main reason I opted for this was because the order is important for the files, and it would be harder to read what is parsed where, as all the lines would not be aligned anymore. But I agree that the implicit dependency on the order is not ideal.

Reply:
Another problem is that we only pass the dictionary here, and not the model itself, thus all the variables would then be defined as `data["property-name"]` which I'm not super happy about either.